boat_simulator/nodes/physics_engine/fluid_generation.py

```diff
@@ -0,0 +1,37 @@
+"""Generator for fluid vectors used in the physics engine."""
+
+import numpy as np
+from numpy.typing import NDArray
+
+from boat_simulator.common.generators import VectorGenerator
+from boat_simulator.common.types import Scalar
+
+
+class FluidGenerator:
+    def __init__(self, generator: VectorGenerator):
+        self.__generator = generator
+        self.__velocity = np.array([0, 0])
+        self.__speed = 0
+        self.__direction = 0
+
+    def next(self) -> NDArray:
+        """Updates the fluid simulator by generating the next velocity vector.
+
+        Returns:
+            NDArray: An array representing the updated velocity vector for the fluid simulation.
+        """
+
+        # TODO: Implement this method
+        return np.array([0, 0])
+
+    @property
+    def velocity(self) -> NDArray:
+        return self.__velocity
+
+    @property
+    def speed(self) -> Scalar:
+        return self.__speed
+
+    @property
+    def direction(self) -> Scalar:
+        return self.__direction
```

Comment:
These can probably contain logic to directly compute these values given the velocity vector. Doing it this way means that we need to manually update `self.__speed` and `self.__direction` every time the velocity is updated.

Remember to specify the range and unit for the direction, speed, and velocity